astropy/units/typing.py

```diff
@@ -2,20 +2,40 @@
 
 from __future__ import annotations
 
-__all__ = ["QuantityLike", "UnitPower", "UnitPowerLike", "UnitScale", "UnitScaleLike"]
+__all__ = [
+    "EquivalencyPair",
+    "QuantityLike",
+    "UnitPower",
+    "UnitPowerLike",
+    "UnitScale",
+    "UnitScaleLike",
+]
 
 
+from collections.abc import Callable
 from fractions import Fraction
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 import numpy as np
 import numpy.typing as npt
 
-from astropy.units import Quantity
+from astropy.units import Quantity, UnitBase
 
 if TYPE_CHECKING:
     from typing import TypeAlias
 
+EquivalencyPair: TypeAlias = (
```

Comment:
This name is based on the phrasing used in our documentation. For example, https://github.com/astropy/astropy/blob/92b38aebbba6422890f1023031558f6df738e2a5/docs/units/equivalencies.rst?plain=1#L16 https://github.com/astropy/astropy/blob/92b38aebbba6422890f1023031558f6df738e2a5/astropy/units/equivalencies.py#L107-L109 or https://github.com/astropy/astropy/blob/92b38aebbba6422890f1023031558f6df738e2a5/astropy/units/equivalencies.py#L598-L600